lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/NettyMessagingTransport.java

```diff
@@ -130,10 +132,25 @@ private NettyMessagingTransport(
 
     final String host = UNKNOWN_HOST_NAME.equals(hostAddress) ? localAddressProvider.getLocalAddress() : hostAddress;
 
+    //TODO[JIRA REEF-1871] Implement HTTPS with sslContext.
+
+    // for HTTP and default Netty
+    if (protocolType == ProtocolType.HTTP) {
+      this.uri = URI.create("http://" + hostAddress);
```

Comment:
@nhne `hostAddress` -> `host`
I've got an exception message when I tested it on my local machine, even though the test passed  
```
Caused by: java.lang.IllegalArgumentException: Illegal character in fragment at index 8: http://##UNKNOWN##
	at java.net.URI.create(URI.java:852)
	at org.apache.reef.wake.remote.transport.netty.NettyMessagingTransport.<init>(NettyMessagingTransport.java:139)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at org.apache.reef.tang.implementation.java.InjectorImpl.injectFromPlan(InjectorImpl.java:637)
	... 41 more
```

Reply:
Sorry for late check. I had to reconfigure my environment to reproduce the error. I'll fix it and push a commit now!